src/utils/download.ts

```diff
@@ -0,0 +1,25 @@
+import * as fs from 'fs';
+import path from 'path';
+import axios, { AxiosResponse } from 'axios';
+
+/**
+ * Downloads file from remote HTTP[S] host and puts its contents to the
+ * specified location.
+ * @param url URL of the file to download.
+ * @param destinationPath Path to the destination file.
+ */
+export async function download(url: string, filePath: string): Promise<string> {
+	const fileName = url.split('/').pop();
+	if (!fileName) {
+		throw new Error('Invalid remote path. No file name found.');
+	}
+	try {
+		const response: AxiosResponse = await axios.get(url, { responseType: 'arraybuffer' });
+		const pathToFile = path.join(filePath, fileName);
+
+		fs.writeFileSync(pathToFile, response.data);
```

Comment:
As mentioned above, I'm worried about the scalability of this approach (i.e. reading the whole file into memory). I'd prefer if we streamed the data to disk. `pipeline` is a pretty clean way to do that that deals with a lot of error handling for you. See this SO: https://stackoverflow.com/a/64925465/1723537